### ISSUE TITLE
[MNG-4310] Can't use property names in resources element

### ISSUE THREAD

**Issue by jira-importer** [apache/maven#3821]
**[Zarar Siddiqi](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=zarar)** opened **[MNG-4310](https://issues.apache.org/jira/browse/MNG-4310?redirect=false)** and commented

```xml
<build>
    <resources>
         <resource>
            <directory>target/gen/doclet/WEB-INF/classes/META-INF</directory>
            <filtering>true</filtering>
            <targetPath>META-INF</targetPath>
         </resource>            
    </resources>
</build>
```

I can't use `${project.build.directory`} instead of _target_ in the _directory_ element in the above snippet.  Same is true for other Maven and project properties.

---

**Issue Links:**
- [MNG-1906](https://issues.apache.org/jira/browse/MNG-1906) properties not being replaced before being passed to plugins (specificaly the surefire plugin)
 (_**"duplicates"**_)
- [MNG-1486](https://issues.apache.org/jira/browse/MNG-1486) Can't use pom properties inside resource directory  tag
 (_**"duplicates"**_)
- [MRESOURCES-55](https://issues.apache.org/jira/browse/MRESOURCES-55) filter file location does not expand ${project.build.directory}
 (_**"is duplicated by"**_)

8 votes, 6 watchers

**Comment by jira-importer** [comment]
**[julien soula](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jsoula)** commented

The pb is on the maven core, before values are passed to the plugin. Now it is the role of maven to resolved dollar brace properties or the role of the plugin, especially in core plugins like maven-resources-plugin?

**Comment by jira-importer** [comment]
**[John Dennis Casey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jdcasey)** commented

need to verify that this is fixed (should have been fixed in either 2.1.0-M1 or 2.1.0).

**Comment by jira-importer** [comment]
**[Brett Porter](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=brettporter)** commented

works in 2.2.1
